src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts

```diff
@@ -0,0 +1,261 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from "chai";
+import { MANIFESTS } from "../../manifests";
+import { TABLE_MANIFEST } from "../../manifests/table/table";
+import { Essence } from "../../models/essence/essence";
+import { SeriesList } from "../../models/series-list/series-list";
+import { PERCENT_FORMAT } from "../../models/series/series-format";
+import { SeriesFixtures } from "../../models/series/series.fixtures";
+import { SortDirection } from "../../models/sort/sort";
+import { SplitFixtures } from "../../models/split/split.fixtures";
+import { Splits } from "../../models/splits/splits";
+import { dataCube } from "../test/data-cube.fixture";
+import { essence } from "../test/essence.fixture";
+import { count, quantile, sum } from "../test/measure";
+import { SeriesDefinitionFixtures } from "./series-definition.fixtures";
+import { SplitDefinitionFixtures } from "./split-definition.fixtures";
+import { ViewDefinition4 } from "./view-definition-4";
+import { viewDefinition } from "./view-definition-4.fixture";
+import { ViewDefinitionConverter4 } from "./view-definition-converter-4";
+
+const converter = new ViewDefinitionConverter4();
+const toEssence = (viewDef: ViewDefinition4) => converter.fromViewDefinition(viewDef, dataCube, MANIFESTS);
+
+function assertEqlEssence(actual: Essence, expected: Essence) {
+  try {
+    expect(actual.equals(expected)).to.be.true;
+  } catch (e) {
+    expect(actual.toJS()).to.deep.equal(expected.toJS());
+    throw e;
+  }
+}
+
+function assertEqlEssenceWithoutVisResolve(actual: Essence, expected: Essence) {
+  const actualWithoutVisResolve = actual.set("visResolve", null);
+  const expectedWithoutVisResolve = expected.set("visResolve", null);
+  try {
+    expect(actualWithoutVisResolve.equals(expectedWithoutVisResolve)).to.be.true;
+  } catch (e) {
+    expect(actualWithoutVisResolve.toJS()).to.deep.equal(expectedWithoutVisResolve.toJS());
+    throw e;
+  }
+}
+
+describe("ViewDefinitionConverter4", () => {
+  describe("Base case", () => {
+    it("converts to default essence", () => {
+      const result = toEssence(viewDefinition());
+      const expected = essence();
+      assertEqlEssence(result, expected);
+    });
+  });
+
+  describe("Splits", () => {
+    it("reads string split", () => {
+      const result = toEssence(viewDefinition({
+        splits: [SplitDefinitionFixtures.stringSplitDefinition("string_a", "string_a")],
```

Comment:
I'd hide the boilerplate like `SplitDefinitionFixtures.stringSplitDefinition` and `Splits.fromSplit` in some local helper. There is a lot of noise in these tests.